realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -112,6 +112,7 @@ public boolean handleMessage(Message message) {
                 }
                 case COMPLETED_UPDATE_ASYNC_QUERIES: {
                     // this is called once the background thread completed the update of the async queries
+                    notifyAllListeners();
```

Comment:
I'm not sure about this either.
The current implementation of `completedAsyncQueriesUpdate` will notify the listeners if we advance the Realm.
```java
// notify listeners only when we advanced
            if (compare != 0) {
                notifyGlobalListeners();
                // notify RealmResults & RealmObject callbacks (type based notifications)
                notifySyncRealmResultsCallbacks();
                notifyRealmObjectCallbacks();
            }
```
maybe you can refactor it to 
```java
 if (compare != 0) {
    notifyAllListeners();
  }
```